### DISCUSSION TITLE
getStaticPaths, fetch to api call with current hostname

### DISCUSSION THREAD

**Discussion by MikeDrenth** [vercel/next.js#44800] [Help]
Hello,

To start off, i am using this template: https://github.com/vercel/platforms

We are trying to build a SaaS project with the previous mentioned project. That means that, we use this repo to build multiple websites on one codebase. 

Now the issue i am facing is that, getStaticPaths is trying to get generate all the paths, which is cool and i understand why it should. But the problem i am having is that, i need to make a API call with a required parameter that is the current domain. Then it will give me a list of all the current pages.

The problem is that, i need a way to pass that into my call ` getSiteInfo(site as string)` but at this point i am unable to find the current domain / url inside getStaticPaths, because i don't have access to NextRequest or NextResponse. So the question is, is there any way i can send the required domain/website url into the request inside getStatisPaths?

**Comment by icyJoseph** [comment]
This is a problem with white-label applications.

I think one way out is to provide the expected domain as an ENV VAR, of course that has downsides too, but if you need it, then the options are pretty much limited.

The problem is that these run during build time or server, without any knowledge of the environment they are running inside of.
